File: src/main/java/to/joeli/jass/client/strategy/training/DataSet.java
package to.joeli.jass.client.strategy.training;

import java.util.List;

public abstract class DataSet {
	private static final String BASE_PATH = "src/main/resources/";

	public abstract List getFeatures();

	public abstract List getTargets();

	protected abstract String getNetworkTypePath();

	public String getPath(TrainMode trainMode) {
		return BASE_PATH + trainMode.getPath() + getNetworkTypePath();
	}

	public String getFeaturesPath(TrainMode trainMode) {
		return getPath(trainMode) + "features/";
	}

	public String getTargetsPath(TrainMode trainMode) {
		return getPath(trainMode) + "targets/";
	}
}

File: src/main/java/to/joeli/jass/client/strategy/training/DataSet.kt
package to.joeli.jass.client.strategy.training


import java.util.*

class DataSet(private val cardsFeaturesQueue: Queue<Array<DoubleArray>>,
              private val scoreFeaturesQueue: Queue<Array<DoubleArray>>,
              private val cardsTargetsQueue: Queue<Array<IntArray>>,
              private val scoreTargetsQueue: Queue<Double>) {

    val cardsFeatures: List<Array<DoubleArray>>
        get() = ArrayList(cardsFeaturesQueue)

    val scoreFeatures: List<Array<DoubleArray>>
        get() = ArrayList(scoreFeaturesQueue)

    val cardsTargets: List<Array<IntArray>>
        get() = ArrayList(cardsTargetsQueue)

    val scoreTargets: List<Double>
        get() = ArrayList(scoreTargetsQueue)
}
